flow/layers/color_filter_layer.cc

```diff
@@ -54,18 +62,23 @@ void ColorFilterLayer::Preroll(PrerollContext* context) {
   // modulation so in that case we can apply the opacity on behalf of our
   // ancestors - otherwise we can apply no attributes.
   if (filter_) {
+    set_paint_bounds(child_bounds);
+
     context->renderable_state_flags =
         filter_->can_commute_with_opacity()
             ? LayerStateStack::kCallerCanApplyOpacity
             : 0;
+  } else {
+    // else - we can apply whatever our children can apply.
+    set_paint_bounds(child_bounds);
   }
-  // else - we can apply whatever our children can apply.
 }
 
 void ColorFilterLayer::Paint(PaintContext& context) const {
   FML_DCHECK(needs_painting(context));
 
   auto mutator = context.state_stack.save();
+  mutator.translate(offset_);
```

Comment:
There is a bug in IFL that we are currently fixing where this line here causes the offset to be applied doubly when caching children. See https://github.com/flutter/engine/pull/38567 on our discussion about how the interplay of offset/integralTransform/caching should play out.